instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,311 @@
+    }
+
+    @Test
+    fun testBillingSessionsForStoppedAndResumedActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(3, billingRequests.size)
+
+        assertTrue(billingRequests[0].isActiveGuidanceSession)
+        assertSessionDuration(0, billingRequests[0].duration)
+        assertTrue(billingRequests[1].isFreeDriveSession)
+        assertSessionDuration(SESSION_PAUSE_DURATION_SECONDS, billingRequests[1].duration)
+        assertTrue(billingRequests[2].isActiveGuidanceSession)
+        assertSessionDuration(0, billingRequests[2].duration)
+    }
+
+    private companion object {
+
+        const val HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS = 500L
+        const val SESSION_DURATION_SECONDS = 1L
+        const val SESSION_PAUSE_DURATION_SECONDS = 1L
+        const val ACTIVE_GUIDANCE_SKU_PREFIX = "10a"
+        const val FREE_DRIVE_SKU_PREFIX = "10b"
+
+        const val SESSIONS_ENDPOINT_PATH = "/sdk-sessions"
+
+        val HttpServiceEvent.isSessionEvent: Boolean
+            get() = url.path.startsWith(SESSIONS_ENDPOINT_PATH)
+
+        val HttpServiceEvent.skuParameter: String?
+            get() = url.parameters()["sku"]
+
+        val HttpServiceEvent.isActiveGuidanceSession: Boolean
+            get() = skuParameter?.startsWith(ACTIVE_GUIDANCE_SKU_PREFIX) == true
+
+        val HttpServiceEvent.isFreeDriveSession: Boolean
+            get() = skuParameter?.startsWith(FREE_DRIVE_SKU_PREFIX) == true
+
+        val HttpServiceEvent.duration: Long?
+            get() = url.parameters()["duration"]?.toLong()
+
+        fun HttpServiceEventsObserver.billingRequests(): List<HttpServiceEvent.Request> {
+            return onRequestEvents
+                .filter {
+                    it.isSessionEvent && (it.isFreeDriveSession || it.isActiveGuidanceSession)
+                }
+        }
+
+        private fun assertSessionDuration(expectedDurationSeconds: Long?, duration: Long?) {
+            if (expectedDurationSeconds == null) {
+                assertNull(duration)
+                return
+            }
+
+            assertNotNull("Session duration was null", duration)
+            requireNotNull(duration)
+
+            // Allow at most one second for session time calculation error
+            val approxSame = (duration >= expectedDurationSeconds) &&
```

Comment:
Why aren't you using abs here? Do we only allow the actual duration to be greater than expected?

Reply:
If actual duration is less than expected it's a bug because duration can't be less than time elapsed between session start and stop. That's why `abs` doesn't make sense here.

Reply:
Agreed to add a separate check for a case when duration is less than expected